lang/cs/Org.Apache.REEF.Client/Multi/MultiRuntimeConfigurationBuilder.cs

```diff
@@ -0,0 +1,249 @@
+        {
+            this.hdiPassword = pwd;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetHDInsightUrl(string hdiUrl)
+        {
+            this.hdiUrl = hdiUrl;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetAzureBlobContainerName(string containerName)
+        {
+            this.jobSubmissionDirectoryPrefix = containerName;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetAzureBlobConnectionString(string connectionString)
+        {
+            this.connectionString = connectionString;
+            return this;
+        }
+
+        public IConfiguration Build()
+        {
+            if (!this.submissionRunitme.HasValue)
+            {
+                throw new ArgumentException("Submission runntime was not set");
+            }
+
+            if (!this.defaultRuntime.IsPresent() && this.runtimes.Count == 1)
+            {
+                this.defaultRuntime = Optional<RuntimeName?>.Of(this.runtimes.First());
+            }
+
+            if (!this.defaultRuntime.IsPresent())
+            {
+                throw new ArgumentException("Default runntime was not set");
+            }
+
+            if (this.runtimes.Contains(RuntimeName.Local) && !this.maxEvaluatorsNumber.HasValue)
+            {
+                throw new ArgumentException("Max evaluators number should be a positive integer");
+            }
+
+            IList<string> runtimes = this.runtimes.Select(x => x.ToString()).ToList();
+            var config = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindNamedParameter<DefaultRuntime, string>(GenericType<DefaultRuntime>.Class, this.defaultRuntime.Value.ToString());
+
+            foreach (var runtime in runtimes)
+            {
+                config.BindSetEntry<DefinedRuntimes, string>(GenericType<DefinedRuntimes>.Class, runtime);
+            }
+                
+            AvroConfigurationSerializer serializer = new AvroConfigurationSerializer();
+            var serializedConfig = serializer.ToString(config.Build());
+            var configBuilder = TangFactory.GetTang().NewConfigurationBuilder().BindImplementation(
+                    GenericType<IYarnREEFDotNetAppParamsSerializer>.Class,
+                    GenericType<YarnMultiREEFDotNetAppParamSerializer>.Class)
+                    .BindNamedParameter<LauncherClassName, string>(
+                        GenericType<LauncherClassName>.Class,
+                        MultiRuntimeLauncher)
+                    .BindNamedParameter<AdditionalDriverAppConfiguration, string>(
+                        GenericType<AdditionalDriverAppConfiguration>.Class,
+                        serializedConfig);
+           
+            IConfiguration ret = null;
+            switch (this.submissionRunitme.Value)
+            {
+                // add submission runtime 
+                case SubmissionRuntimeName.HDInsight:
```

Comment:
What is the difference between `SubmissionRuntimeName` and `RuntimeName`? What is the significance of `defaultRuntime`? Why can't the `defaultRuntime` be the `submissionRuntime`?